tests/moe_ep/test_mega_workspace_api.py

```
@@ -0,0 +1,569 @@
+            side_effect=lambda fp: ("workspace-api", fp.max_tokens_per_rank),
+        ),
+        mock.patch.object(
+            layer._kernel,
+            "_allocate_workspace",
+            side_effect=allocate,
+        ),
+    ):
+        first = layer.create_workspace(128)
+        with pytest.raises(
+            MoEEpConfigError,
+            match="workspace profile.*already exists",
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)